be/src/fs/fs_s3.cpp

```diff
@@ -149,39 +160,48 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const ClientConfigurati
 }
 
 static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri, const FSOptions& opts) {
-    Aws::Client::ClientConfiguration config = S3ClientFactory::getClientConfig();
+    S3ClientFactory::ClientConfiguration config = S3ClientFactory::getClientConfig();
+    config.bucket = uri.bucket();
+
+    Aws::Client::ClientConfiguration& client_config = config.client_config;
+
     const THdfsProperties* hdfs_properties = opts.hdfs_properties();
     if (hdfs_properties != nullptr) {
         DCHECK(hdfs_properties->__isset.end_point);
         if (hdfs_properties->__isset.end_point) {
-            config.endpointOverride = hdfs_properties->end_point;
+            client_config.endpointOverride = hdfs_properties->end_point;
         }
         if (hdfs_properties->__isset.ssl_enable && hdfs_properties->ssl_enable) {
-            config.scheme = Aws::Http::Scheme::HTTPS;
+            client_config.scheme = Aws::Http::Scheme::HTTPS;
         }
 
         if (hdfs_properties->__isset.region) {
-            config.region = hdfs_properties->region;
+            client_config.region = hdfs_properties->region;
         }
         if (hdfs_properties->__isset.max_connection) {
-            config.maxConnections = hdfs_properties->max_connection;
+            client_config.maxConnections = hdfs_properties->max_connection;
         } else {
-            config.maxConnections = config::object_storage_max_connection;
+            client_config.maxConnections = config::object_storage_max_connection;
         }
+        DCHECK(hdfs_properties->__isset.access_key);
+        DCHECK(hdfs_properties->__isset.secret_key);
+        config.access_key_id = hdfs_properties->access_key;
+        config.secret_access_key = hdfs_properties->secret_key;
+
     } else {
         if (!uri.endpoint().empty()) {
-            config.endpointOverride = uri.endpoint();
+            client_config.endpointOverride = uri.endpoint();
         } else if (!config::object_storage_endpoint.empty()) {
-            config.endpointOverride = config::object_storage_endpoint;
+            client_config.endpointOverride = config::object_storage_endpoint;
         } else if (config::object_storage_endpoint_use_https) {
-            config.scheme = Aws::Http::Scheme::HTTPS;
+            client_config.scheme = Aws::Http::Scheme::HTTPS;
         } else {
-            config.scheme = Aws::Http::Scheme::HTTP;
+            client_config.scheme = Aws::Http::Scheme::HTTP;
         }
         if (!config::object_storage_region.empty()) {
-            config.region = config::object_storage_region;
+            client_config.region = config::object_storage_region;
         }
-        config.maxConnections = config::object_storage_max_connection;
+        client_config.maxConnections = config::object_storage_max_connection;
```

Comment:
How many different client configurations in your production?  Because you see in `new_client`, we use linear search & random cache eviction poicy to manage same configured client, and maye you want to consider better cache mechanism.

```
S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const ClientConfiguration& config, const FSOptions& opts) {
    std::lock_guard l(_lock);
    for (size_t i = 0; i < _items; i++) {
        if (_configs[i] == config) return _clients[i];
    }
```

Reply:
won't be too much, we think current simple cache policy is ok.